lib/tasks/server/express-server.js

```diff
@@ -116,7 +116,7 @@ class ExpressServerTask extends Task {
   }
 
   processAppMiddlewares(options) {
-    if (this.project.has(this.serverRoot)) {
+    if (this.project.has(`${this.serverRoot}.js`) || this.project.has([this.serverRoot, 'index'].join(path.sep))) {
```

Comment:
Now this breaks folks using `server/index.ts` along with ts-node.

Would you mind replacing this `if` completely, with something like:

```js
processAppMiddlewares(options) {
  let server;

  try {
    server = this.project.require(this.serverRoot);
  } catch(error) {
	if (error.code === 'MODULE_NOT_FOUND') {
      // do nothing, `server.js`, `server/index.js`, etc were not found
      return;
    }
    throw error;
  }

   // existing logic
}
```

Reply:
it will never support `ts` due to https://github.com/ember-cli/ember-cli/blob/master/lib/models/project.js#L323 implementation

Reply:
> it will never support ts due to https://github.com/ember-cli/ember-cli/blob/master/lib/models/project.js#L323 implementation

Incorrect 😄 

I do agree that that implementation should be fixed (to use `resolve.sync` or something), but the example I gave was using `server/index.ts` and if you look at the existing logic we look for `path.join(this.root, 'server')` and it would exist so we'd never hit the fallback that looks for `path.join(this.root, 'server.js')`.